bundles/binding/org.openhab.binding.powermax/src/main/java/org/openhab/binding/powermax/internal/PowerMaxBinding.java

```diff
@@ -0,0 +1,881 @@
+        //
+
+        if (!connected) {
+            logger.debug("execute(): trying to reconnect...");
+            closeConnection();
+            if (triggerItemsUpdate) {
+                logger.debug("execute(): items update enabled");
+                itemUpdateDisabled = false;
+                triggerItemsUpdate = false;
+            }
+            currentState = new PowerMaxState();
+            openConnection();
+            if (connected) {
+                // TODO is INIT message required or not ?
+
+                if (forceStandardMode) {
+                    currentState.setPowerlinkMode(false);
+                    updateItemsFronAlarmState(PowerMaxSelectorType.PANEL_MODE, currentState);
+                    settings = PowerMaxPanelSettings.getThePanelSettings();
+                    settings.process(false, panelType, null);
+                    updateItemsFronPanelSettings();
+                    logger.info("PowerMax alarm binding: running in Standard mode");
+                    PowerMaxCommDriver.getTheCommDriver().sendMessage(PowerMaxSendType.ZONESNAME);
+                    PowerMaxCommDriver.getTheCommDriver().sendMessage(PowerMaxSendType.ZONESTYPE);
+                    PowerMaxCommDriver.getTheCommDriver().sendMessage(PowerMaxSendType.STATUS);
+                } else {
+                    PowerMaxCommDriver.getTheCommDriver().startDownload();
+                }
+            } else {
+                logger.debug("execute(): reconnection failed");
+            }
+        } else if (triggerItemsUpdate) {
+            logger.debug("execute(): update all items");
+            itemUpdateDisabled = false;
+            triggerItemsUpdate = false;
+            // Adjust all the items to the current alarm state
+            updateItemsFronAlarmState(currentState);
+            // Adjust all the items to the current alarm settings
+            updateItemsFronPanelSettings();
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void bindingChanged(BindingProvider provider, String itemName) {
+        logger.debug("bindingChanged(): Item Name: {}", itemName);
+        logger.debug("Provider items count: {}", provider.getItemNames().size());
+
+        itemUpdateDisabled = true;
+        if (provider.getItemNames().size() >= itemCount) {
+            triggerItemsUpdate = true;
+        }
+        itemCount = provider.getItemNames().size();
+
+        super.bindingChanged(provider, itemName);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void allBindingsChanged(BindingProvider provider) {
+        logger.debug("allBindingsChanged()");
+        logger.debug("Provider items count: {}", provider.getItemNames().size());
+
+        itemUpdateDisabled = true;
+        if (provider.getItemNames().size() > 0) {
+            triggerItemsUpdate = true;
+        }
+        itemCount = provider.getItemNames().size();
+
+        super.allBindingsChanged(provider);
+    }
```

Comment:
Yes and that works well ;)
The refresh of items is done at the next refresh cycle (max 20 seconds later).